frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx

```diff
@@ -85,34 +85,33 @@ class Visualization extends PureComponent {
     computedSettings: {},
   };
 
-  UNSAFE_componentWillMount() {
-    this.transform(this.props);
+  constructor(props) {
+    super(props);
+    this.state = this.transformPropsToState(props);
   }
```

Comment:
Use constructor instead of UNSAFE_componentWillMount to compute and load the initial state